### DISCUSSION TITLE
More customizable /list command

### DISCUSSION THREAD

**Discussion by zRiaz** [EssentialsX/Essentials#29] [Ideas and Feature Suggestions]
The next feature I am requesting is a better /list.
Currently its very limited to how customized it can become, if you could make something like this:

"------------------------------------------------------------------"

Staff Online: (staff)
Total Members online: x/x

"--------------------------------------------------------------------"

Or whatever how customized you want it, that would be a lot better. I know there are plugins that does both of these things, but I mean Essentials already have them just that its not so good implemented in Essentials. If you could add these then those plugins wouldnt be nessecerry.

**Comment by Kakifrucht** [comment]
This is already possible (kinda).

**Comment by zRiaz** [comment]
I attempted customizing it with the language file, but it didnt turn out so well..

I would like to see this implemented as an easy way to configure with a few variables in the list section in the config were it would be possible to make the list appear pretty much however you want.

I also would imagine it isnt too much of a big task to do, neither that it would make any difference in how Essentials works, basically just an improvement. So yea thought I would suggest to have this added.

(Which also would mean I would get 1 less plugin ;) )

**Comment by Kakifrucht** [comment]
I think you simply missed this being already implemented. There is no playing around with the language file.

Config file only:
Sort output of /list command by groups.
You can hide and merge the groups displayed in /list by defining the desired behaviour here.
Detailed instructions and examples can be found on the wiki: http://wiki.ess3.net/wiki/List

Example:

list:
    Staff: owner admin moderator
    Donatros: vip premium ultra
    Players: players

**Comment by zRiaz** [comment]
But its not customizeable.... ill give examples tomorrow.

**Comment by zRiaz** [comment]
Customizeable list: http://gyazo.com/968926865c6165016a5e791f86b7a07e
Essentials list: http://gyazo.com/3ef63099ecf69bec3e9472c55b6e97f5

**Comment by zRiaz** [comment]
Would still love to see this implemented. Instead of having to use external plugins when Essentials have a almost similiar feature which could just so easily be improved to this..

**Comment by DrewAMitchell** [comment]
Could potentially done with a String list as a .txt file?

**Comment by Flavourized** [comment]
> However, some servers (like mine) have an entire rank up system where players can have multiple groups. Their primary group probably isnt the group that /list is looking for.

An example for this would be players who have a staff rank as well as a donator rank and a main rank. The /list will be looking for the staff ranks, but the players primary group is his main rank, so he is skipped from showing up on /list.

I want to say make it where /list is looking for a permission, maybe essentials.list.. This permission would be added to the groups, and then /list would check if the player has this permission. This will solve the need for changing primary group.

And then (probably goes w/o saying), keep the customtext option so we can customize how we want the command to look.

**Comment by mdcfe** [comment]
If this is something that people are interested in implementing, I think this needs further discussion and input so we can agree on what changes should be made.
